app/src/main/java/io/appium/uiautomator2/model/internal/CustomUiDevice.java

```diff
@@ -94,8 +94,26 @@ public UiAutomation getUiAutomation() {
         return getInstrumentation().getUiAutomation();
     }
 
-    private UiObject2 toUiObject2(Object selector, AccessibilityNodeInfo node) {
-        Object[] constructorParams = {getUiDevice(), selector, node};
+    private UiObject2 toUiObject2(@Nullable Object selector, @Nullable AccessibilityNodeInfo node) {
+        if (selector == null) {
+            // FIXME: The 'selector' should be proper By instance as non-null in the interaction.
+            Logger.debug("FIXME: selector argument should not be null in androidx.test.uiautomator:uiautomator:2.3.0");
+        }
+
+        // TODO: remove this comment after upgrading to androidx.test.uiautomator:uiautomator:2.3.0
+        // UiObject2 with androidx.test.uiautomator:uiautomator:2.3.0 has below code to crate the instance,
+        // thus if the node was None, it should create an empty element for the AccessibilityNodeInfo.
+        //    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+        //        AccessibilityWindowInfo window = UiObject2.Api21Impl.getWindow(cachedNode);
+        //        mDisplayId = window == null ? Display.DEFAULT_DISPLAY : UiObject2.Api30Impl.getDisplayId(window);
+        //    } else {
+        //        mDisplayId = Display.DEFAULT_DISPLAY;
+        //    }
+        AccessibilityNodeInfo accessibilityNodeInfo =
+                (node == null && android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.R)
+                        ?  new AccessibilityNodeInfo()
```

Comment:
extra space here

Reply:
Removed https://github.com/appium/appium-uiautomator2-server/pull/589/commits/cd5692b8119e9c5dbc923813d5e5d4e4525f2218
I used AS's config so it added these spaces automatically. I guess we could modify config file etc?